packages/llamaindex/src/storage/vectorStore/UpstashVectorStore.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  VectorStoreBase,
+  type IEmbedModel,
+  type MetadataFilters,
+  type VectorStoreNoEmbedModel,
+  type VectorStoreQuery,
+  type VectorStoreQueryResult,
+} from "./types.js";
+
+import type { BaseNode, Metadata, TextNode } from "@llamaindex/core/schema";
+import { getEnv } from "@llamaindex/env";
+import { Index } from "@upstash/vector";
+import { metadataDictToNode, nodeToMetadata } from "./utils.js";
+
+type UpstashParams = {
+  namespace?: string;
+  token: string;
+  endpoint: string;
+} & IEmbedModel;
+
+/**
+ * Provides support for writing and querying vector data in Upstash.
+ */
+export class UpstashVectorStore
+  extends VectorStoreBase
+  implements VectorStoreNoEmbedModel
+{
+  storesText: boolean = true;
+
+  private db: Index;
+  namespace: string;
+
+  constructor(params?: UpstashParams) {
+    super(params?.embedModel);
+    this.namespace = params?.namespace ?? "";
+    const token = params?.token ?? getEnv("UPSTASH_VECTOR_REST_TOKEN");
+    const endpoint = params?.endpoint ?? getEnv("UPSTASH_VECTOR_REST_URL");
+
+    if (!token) {
+      throw new Error(
+        "Must specify UPSTASH_VECTOR_REST_TOKEN via env variable.",
+      );
+    }
+    if (!endpoint) {
+      throw new Error("Must specify UPSTASH_VECTOR_REST_URL via env variable.");
+    }
+    this.db = new Index({ token, url: endpoint });
+  }
+
+  private async getDb(): Promise<Index> {
```

Comment:
do we really need this method? we can achieve the same thing with`new UpstashVectorStore`

Reply:
It is in the interface of the class